fe/fe-core/src/main/java/org/apache/doris/qe/MysqlConnectProcessor.java

```diff
@@ -81,35 +85,12 @@ private void debugPacket() {
         }
     }
 
-    // process COM_EXECUTE, parse binary row data
-    // https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_com_stmt_execute.html
-    private void handleExecute() {
-        // debugPacket();
-        packetBuf = packetBuf.order(ByteOrder.LITTLE_ENDIAN);
-        // parse stmt_id, flags, params
-        int stmtId = packetBuf.getInt();
-        // flag
-        packetBuf.get();
-        // iteration_count always 1,
-        packetBuf.getInt();
-        if (LOG.isDebugEnabled()) {
-            LOG.debug("execute prepared statement {}", stmtId);
-        }
-        PrepareStmtContext prepareCtx = ctx.getPreparedStmt(String.valueOf(stmtId));
-        if (prepareCtx == null) {
-            if (LOG.isDebugEnabled()) {
-                LOG.debug("No such statement in context, stmtId:{}", stmtId);
-            }
-            ctx.getState().setError(ErrorCode.ERR_UNKNOWN_COM_ERROR,
-                    "msg: Not supported such prepared statement");
-            return;
-        }
-        ctx.setStartTime();
-        if (prepareCtx.stmt.getInnerStmt() instanceof QueryStmt) {
+    private void handleExecute(PrepareStmt prepareStmt, long stmtId) {
```

Comment:
all prepare and execute code should be put into PrepareCommand#run and ExecuteCommand#run

Reply:
placeholders  should be handled and parsed before  ExecuteCommand#run

Reply:
we could do this in executeCommand#run by using ExpressionRewriter